Tools/debug/CollectSystemInfo.ps1

```diff
@@ -0,0 +1,248 @@
+
+function Get-InstalledApplications {
+    try {
+        Get-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall\*' |
+        Select-Object -Property DisplayName, DisplayVersion, Publisher, InstallDate |
+        Export-Csv -Path InstalledApplications.csv -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of installed applications: $_"
+    }
+}
+
+function Get-InstalledKBs {
+    try {
+        Get-HotFix | Select-Object -Property Description, HotFixID, InstalledOn | Export-Csv -Path InstalledKBs.csv -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of installed KBs: $_"
+    }
+}
+
+function Get-WindowsMemoryDump {
+    if (Test-Path -Path "$env:SystemRoot\MEMORY.DMP") {
+        try {
+            $zipFile = ".\MEMORY.zip"
+            Compress-Archive -Path "$env:SystemRoot\MEMORY.DMP" -DestinationPath $zipFile -Force
+        } catch {
+            Write-Warning "Failed to collect memory dump file: $_"
+        }
+    } else {
+        Write-Host "No MEMORY.DMP file found."
+    }
+    
+    $minidumpPath = "$env:SystemRoot\Minidump"
+    if (Test-Path -Path $minidumpPath) {
+        try {
+            $minidumpFiles = Get-ChildItem -Path $minidumpPath -Filter "*.dmp"
+            if ($minidumpFiles.Count -gt 0) {
+                $minidumpZipFile = ".\Minidumps.zip"
+                Compress-Archive -Path $minidumpFiles.FullName -DestinationPath $minidumpZipFile -Force
+            } else {
+                Write-Host "No minidump files found in $minidumpPath."
+            }
+        } catch {
+            Write-Warning "Failed to collect minidump files: $_"
+        }
+    } else {
+        Write-Host "No Minidump directory found."
+    }
+}
+
+function Remove-TempFiles {
+    param (
+        [string[]]$FilesToDelete
+    )
+    foreach ($file in $FilesToDelete) {
+        $resolvedFiles = Resolve-Path $file -ErrorAction SilentlyContinue
+        foreach ($resolvedFile in $resolvedFiles) {
+            try {
+                Remove-Item -Path $resolvedFile -Recurse -Force
+            } catch {
+                Write-Warning "Failed to delete file: $resolvedFile - $_"
+            }
+        }
+    }
+}
+
+function Write-InformationToArchive {
+    try {
+        $date = Get-Date -Format "yyyy-MM-dd_HH-mm-ss"
+        $archiveFile = "SystemInfo_$date.zip"
+        $filesToArchive = @('*.txt', '*.csv', '*.evtx', '*.log')
+        if (Test-Path -Path "MEMORY.DMP") {
+            $filesToArchive += 'MEMORY.DMP'
+        }
+        if (Test-Path -Path "LocaleMetaData") {
+            $filesToArchive += 'LocaleMetaData'
+        }
+        Compress-Archive -Path $filesToArchive -DestinationPath $archiveFile -Force
+        Remove-TempFiles $filesToArchive
+        Write-Host "Archiving completed ($archiveFile)."
+    } catch {
+        Write-Warning "Failed to archive information: $_"
+    }
+}
+
+Write-Host "Starting system information collection..."
```

Comment:
Remove the start and completion messages. It's obvious that it starts collecting information after invoking the script, and you should be always greeted with the prompt after it finishes.